x-pack/plugins/apm/server/routes/alerts/rule_types/error_count/register_error_count_rule_type.ts

```diff
@@ -149,40 +146,42 @@ export function registerErrorCountRuleType({
 
         const errorCountResults =
           response.aggregations?.error_counts.buckets.map((bucket) => {
-            const [serviceName, environment] = bucket.key;
+            const groupByFields = bucket.key.reduce(
+              (obj, bucketKey, bucketIndex) => {
+                obj[allGroupbyFields[bucketIndex]] = bucketKey;
```

Comment:
I'm wary of using the index here. Are we enforcing the order somewhere?

Reply:
The `bucket.key` contains ONLY the values in the order the group by fields are added in the rule. It doesn't have the field names (like `service.name`: `opbeans-go`), so we need to reply on the index to generate this mapping.

Reply:
The multi_terms agg doesn't return it, but you're telling the multi_terms agg the terms (and the order), so you should be able to map the field names to the bucket key.

Reply:
Yes, that is what I am doing here with `allGroupbyFields` which is also used in mulit_terms agg.